src/iqm/cirq_iqm/iqm_operation_mapping.py

```diff
@@ -123,4 +122,25 @@ def map_operation(operation: Operation) -> Instruction:
             args={},
         )
 
+    if isinstance(operation, ClassicallyControlledOperation):
+        if len(operation._conditions) > 1:
+            raise OperationNotSupportedError('Classically controlled gates can currently only have one condition.')
+        if isinstance(operation._conditions[0], (SympyCondition)):
+            raise OperationNotSupportedError('SympyConditions are not natively supported')
```

Comment:
I think both measurement keys as strings and MeasurementKey instances are converted into a `cirq.KeyCondition`, so it's safer to check for those, and raise an error if we see anything else.

Reply:
```suggestion
        if not isinstance(operation._conditions[0], KeyCondition):
            raise OperationNotSupportedError('Only KeyConditions are supported in classically controlled operations.')
```